components/lifesensor_macro/include/macro_queue.h

```diff
@@ -0,0 +1,64 @@
+#ifndef __MACRO_QUEUE__
+#define __MACRO_QUEUE__
+
+#include <freertos/queue.h>
+
+
+/**
+ * struct lifesensor_queue_t - generic queue struct
+ * @queue:	Holds the FreeRTOS queue.
+ * @handle: Provides a reference to the queue.
+ * @buffer: Stores the address of the queues' buffer.
+ *
+ * This struct serves as an abstraction for operating with FreeRTOS queues such
+ * that it includes necessary fields for managing a queues data. It is intended
+ * to be used with the LIFESENSOR_QUEUE() macro.
+ */
+typedef struct {
+	StaticQueue_t queue;
+	QueueHandle_t handle;
```

Comment:
```suggestion
	QueueHandle_t handle;
	StaticQueue_t queue;
```
would move up the handle so the whole struct can be casted directly to the handle